### ISSUE TITLE
consistent clearing of ct outside of segmented structures for multiple cts

### ISSUE THREAD

**Issue by markbangert** [e0404/matRad#297]
In the 4D code we are currently applying the same erasemask for all cts. Therefore the erasemask is build by the union of all segmented structures on _all_ ct phases. This introduces physical inconsistencies in the dose calculation. sometimes the sourrounding of the ct is considered during ray tracing and sometimes not - depending on the very motion pattern. we need to make this clearing with the csts of the individual ct phases, if they are available.

**Comment by eric11210** [comment]
I have "solved" this in my own implementation of 4D dose calculation by not applying the erasemask at all. Most of the voxels have a density close to 0 anyway (except for the couch).

Alternatively, one could apply the deformation vectors to the cst to get a phase-by-phase cst. I am currently doing something similar with the target volume, but it could be easily extended to any other structure.

**Comment by markbangert** [comment]
I have solved this issue by cropping the dose influence in the dev robOptVarRBE branch with information from the cst. . This assumes there is a cst structure available for all phases which might not be the best option. For research purposes however people can easily comment this out. IMO it is better this way as we always set all intensities to zero outside of segmented sturctures in out clinical TPSs. So that way we ensure consistency. @eric11210 I think we'll be in touch soon to ask for your code to deform the cst with vector fields ;)

For the time being however I will close this thread as the problem has been solved.
